### ISSUE TITLE
don't decrypt attachments before tapping + show modal on error

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#1233]
also mostly fixed #1143. The error was that while the text could be decrypted, the attachments were throwing decrypt errors. I've made it recognize decrypt errors and download the original file, but it needs another issue to actually render the decrypt error before downloading, make sure that's what the user wanted.

<img width="366" alt="image" src="https://user-images.githubusercontent.com/6306961/146604031-3ea097af-f8cd-4f33-9d66-626e91c32f36.png">

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/issues/1219#issuecomment-997013348_

**Comment by tomholub** [comment]
Additionally, there is no need to decrypt all attachments when rendering. Decrypt when tapping attachment.

`MessageAttachment` should also have an `isEncrypted: bool`, and we should decide whether to decrypt based on that, when user taps the file.

```swift
                // todo - this is decrypting attachments too early
                //   there is no need to decrypt attachments when rendering message,
                //   since user may never tap them.
                //   Instead decrypt them when actually tapped.
                let decrypted = try await core.decryptFile(
                    encrypted: meta.data,
                    keys: keys,
                    msgPwd: nil
                )
                if let decryptSuccess = decrypted.decryptSuccess {
                    attachment = MessageAttachment(
                        name: decryptSuccess.name,
                        data: decryptSuccess.data
                    )
                } else if let decryptErr = decrypted.decryptErr {
                    // todo - once above todo is done and we are decrypting
                    //   at the time of tapping an attachment, render error modal
                    // if user confirms the modal, they can download the original msg
                    // then can remove the warning
                    logger.logWarning("attachment could not be decrypted due to error: \(decryptErr.error.type) - \(decryptErr.error.message)")
                    attachment = MessageAttachment(
                        name: meta.name,
                        data: meta.data
                    )
                } else {
                    throw AppErr.unexpected("decryptFile: expected one of decryptErr, decryptSuccess to be present")
                }
```

**Comment by tomholub** [comment]
Since we're adding `isEncrypted` to each attachment, we could also surface that in UI / list of attachments, with a lock or similar.

**Comment by tomholub** [comment]
This will also need an appium test for the above message. Tapping the message should produce a modal:

```
Error decrypting attachment: \(decryptErr.error.type) - \(decryptErr.error.message)

This will likely download a corrupted file. Download anyway?

[yes] [no]
§§§
